Dockerfile

```diff
@@ -1,53 +1,66 @@
-ARG GOLANG_BASE
-ARG DEBIAN_BASE
-
-FROM ${GOLANG_BASE} as build
+FROM docker.io/golang:1.18.1-bullseye@sha256:3b1a72af045ad0fff9fe8e00736baae76d70ff51325ac5bb814fe4754044b972 AS build
 
 # tag=1.24.3
 ARG RUSTUP_VERSION=ce5817a94ac372804babe32626ba7fd2d5e1b6ac
 
 # For more information about the snapshots, see: https://snapshot.debian.org/
-RUN echo "\
+RUN printf '\
 deb http://snapshot.debian.org/archive/debian/20220420T025302Z bullseye main\n\
 deb http://snapshot.debian.org/archive/debian/20220420T025302Z bullseye-updates main\n\
 deb http://snapshot.debian.org/archive/debian/20220420T025302Z bullseye-backports main\n\
 deb http://snapshot.debian.org/archive/debian-security/20220420T025302Z bullseye-security main\n\
 deb http://snapshot.debian.org/archive/debian/20220420T025302Z sid main\n\
-" > /etc/apt/sources.list
+' > /etc/apt/sources.list
 
-# NOTICE: -o Acquire::Check-Valid-Until="false" added as a mitigation, see https://github.com/parca-dev/parca-agent/issues/10 for further details.
+# NOTICE: -o Acquire::Check-Valid-Until="false" added as a mitigation,
+# see https://github.com/parca-dev/parca-agent/issues/10 for further details.
+# hadolint ignore=DL3008
 RUN apt-get -o Acquire::Check-Valid-Until="false" update -y && \
-      apt-get install --no-install-recommends -yq llvm-14-dev libclang-14-dev clang-14 make gcc coreutils zlib1g-dev libelf-dev ca-certificates netbase && \
-      ln -s /usr/bin/clang-14 /usr/bin/clang && \
-      ln -s /usr/bin/llc-14 /usr/bin/llc
+    apt-get install --no-install-recommends -yq \
+        llvm-14-dev \
+        libclang-14-dev \
+        clang-14 \
+        make \
+        gcc \
+        coreutils \
+        zlib1g-dev \
+        libelf-dev \
+        ca-certificates \
+        netbase && \
+    ln -s /usr/bin/clang-14 /usr/bin/clang && \
+    ln -s /usr/bin/llc-14 /usr/bin/llc
 
 WORKDIR /parca-agent
 
 # Install Rust
 COPY rust-toolchain.toml /parca-agent
+# SHELL is not supported for OCI image format
+# https://github.com/containers/buildah/blob/v1.26.1/config.go#L366-L377
+# hadolint ignore=DL4006
 RUN curl --proto '=https' --tlsv1.2 -sSf "https://raw.githubusercontent.com/rust-lang/rustup/${RUSTUP_VERSION}/rustup-init.sh" \
     | sh -s -- --default-toolchain none -y
 ENV PATH="/root/.cargo/bin:${PATH}"
 RUN rustup show
 
-ARG ARCH
+ARG TARGETARCH=amd64
```

Comment:
Magical :open_mouth: TIL.

For the ones who also didn't know about it https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/

And Podman supports these ARGs and flags.